consensus/bor/heimdall/client.go

```diff
@@ -198,13 +355,20 @@ func (h *HeimdallClient) FetchMilestoneCount(ctx context.Context) (int64, error)
 		return 0, err
 	}
 
-	ctx = withRequestType(ctx, milestoneCountRequest)
+	ctx = WithRequestType(ctx, MilestoneCountRequest)
 
-	response, err := FetchWithRetry[milestone.MilestoneCountResponse](ctx, h.client, url, h.closeCh)
+	if hmm.IsHeimdallV2 {
```

Comment:
this function uses the same url for resolving milestones count, which is /milestones/count for V2 but for V1 we had /milestone/count